quad_control/src/utilities/utility_functions.py

```diff
@@ -93,146 +79,133 @@ def unit_vec(psi,theta):
 #print unit_vec(0*3.14/180,-90*3.14/180)
 
 
-def bound(x,maxmax,minmin):
-
-    return numpy.maximum(minmin,numpy.minimum(maxmax,x))
 
 #--------------------------------------------------------------------------
 
 #TODO decapitalize: get_euler_angles(rot_max)
-def GetEulerAngles(R):
+def euler_rad_from_rot(R):
 
     #phi   = atan2(R(3,2),R(3,3));
     #theta = asin(-R(3,1));
     #psi   = atan2(R(2,1),R(1,1));
 
-    EULER = numpy.array([0.0,0.0,0.0])
+    euler = np.array([0.0,0.0,0.0])
 
-    EULER[0] = numpy.arctan2(bound(R[2,1],1,-1),bound(R[2,2],1,-1));
-    EULER[1] = numpy.arcsin(-bound(R[2,0],1,-1));
-    EULER[2] = numpy.arctan2(bound(R[1,0],1,-1),bound(R[0,0],1,-1));    
+    euler[0] = np.arctan2(np.clip(R[2,1],-1,1),np.clip(R[2,2],-1,1));
+    euler[1] = np.arcsin(-np.clip(R[2,0],-1,1));
+    euler[2] = np.arctan2(np.clip(R[1,0],-1,1),np.clip(R[0,0],-1,1));    
 
-    return EULER
+    return euler
 
 
-def GetEulerAnglesDeg(R):
-    return GetEulerAngles(R)*180.0/math.pi
+def euler_deg_from_rot(R):
+    return euler_rad_from_rot(R)*180.0/np.pi
 
 
-def GetRotFromEulerAngles(ee_rad):
-    return Rz(ee_rad[2]).dot(Ry(ee_rad[1]).dot(Rx(ee_rad[0])))
+def rot_from_euler_rad(ee_rad):
+    return rot_z(ee_rad[2]).dot(rot_y(ee_rad[1]).dot(rot_x(ee_rad[0])))
 
 
-def GetRotFromEulerAnglesDeg(ee_deg):
-    return GetRotFromEulerAngles(ee_deg*math.pi/180.0)
+def rot_from_euler_deg(ee_deg):
+    return rot_from_euler_rad(ee_deg*np.pi/180.0)
 
 
 
 # testing skew matrix    
-# print skew(numpy.array([1,2,3]))
+# print skew(np.array([1,2,3]))
 
-def quaternion_to_rot(quaternion):
+def rot_from_quaternion(quaternion):
 
     q   = quaternion
     q_v = q[0:3] 
     q_n = q[3]
-    qc  = numpy.concatenate([-q_v,[q_n]])
+    qc  = np.concatenate([-q_v,[q_n]])
 
-    R  = numpy.dot(q,qc)*numpy.identity(3) + 2*q_n*skew(q_v) + 2*numpy.outer(q_v,q_v)
+    R  = np.dot(q,qc)*np.identity(3) + 2*q_n*skew(q_v) + 2*np.outer(q_v,q_v)
 
     return R
 
 
-def quaternion_from_unit_vector(unit_vector,psi):
+def quaternion_from_unit_vector_and_yaw_rad(unit_vector, psi):
 
     r3 = unit_vector
-    r1 = numpy.array([numpy.cos(psi),numpy.sin(psi),0.0])
-    r1 = numpy.dot(numpy.identity(3) - numpy.outer(r3,r3),r1)
-    r1 = r1/numpy.linalg.norm(r1)
-    r2 = numpy.dot(skew(r3),r1)
+    r1 = np.array([np.cos(psi),np.sin(psi),0.0])
+    r1 = np.dot(np.identity(3) - np.outer(r3,r3),r1)
+    r1 = r1/np.linalg.norm(r1)
+    r2 = np.dot(skew(r3),r1)
 
     RR = column_stack((r1,r2,r3))
 
-    return rot_to_quaternion(RR)
```

Comment:
This function was not defined.
I have not defined it either, I have only changed the name.